custom_components/goodwe/switch.py

```diff
@@ -18,72 +18,120 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-LOAD_CONTROL = SwitchEntityDescription(
+
+@dataclass(frozen=True, kw_only=True)
+class GoodweSettingsSwitchEntityDescription(SwitchEntityDescription):
+    """Required values when describing Goodwe switch settings."""
+
+    icon_on: str
+    icon_off: str
+    settings_name: str
+
+
+LOAD_CONTROL = GoodweSettingsSwitchEntityDescription(
     key="load_control",
     name="Load Control",
-    icon="mdi:electric-switch",
+    icon_off="mdi:electric-switch",
+    icon_on="mdi:electric-switch-closed",
     entity_category=EntityCategory.CONFIG,
     device_class=SwitchDeviceClass.OUTLET,
+    settings_name='load_control_switch',
+)
+
+EXPORT_LIMIT_CONTROL = GoodweSettingsSwitchEntityDescription(
+    key="grid_export",
+    name="Grid Export",
+    icon_on="mdi:transmission-tower-off",
+    icon_off="mdi:transmission-tower-import",
```

Comment:
Would not be better `mdi:transmission-tower-export` for on and `mdi:transmission-tower-off` for off?